packages/plugin/src/theia-proposed.d.ts

```diff
@@ -705,3 +610,21 @@ export module '@theia/plugin' {
 
     // #endregion
 }
+
+/**
+ * Thenable is a common denominator between ES6 promises, Q, jquery.Deferred, WinJS.Promise,
+ * and others. This API makes no assumption about what promise library is being used which
+ * enables reusing existing code without migrating to a specific promise implementation. Still,
+ * we recommend the use of native promises which are available in this editor.
+ */
+interface Thenable<T> {
```

Comment:
Isn't this type already defined in `@theia/plugin`? Is it necessary to re-declare it here?

Reply:
I'm open to suggestions. It could be declared only in `theia.proposed.d.ts` and then picked up automatically in `theia.d.ts` but it's good for it to be in `theia.d.ts` for the use of the comparator.